stack/stack.go

```diff
@@ -0,0 +1,189 @@
+	"github.com/aws/aws-sdk-go/service/cloudformation"
+	"github.com/aws/aws-sdk-go/service/sts"
+	"github.com/go-ini/ini"
+	"io/ioutil"
+	"os/user"
+	"time"
+)
+
+const dateFormat = "2006-01-02 15:04:05 MST"
+
+//This function gets template and  name of stack. It creates "CreateStackInput" structure.
+func createStackInput(context *context.Context, template *string, stackName *string) cloudformation.CreateStackInput {
+	templateStruct := cloudformation.CreateStackInput{
+		TemplateBody: template,
+		StackName:    stackName,
+	}
+	return templateStruct
+}
+
+// This function reads "StackName" from Stack in CliArguments and file from TemplatePath in CliArguments. It converts these to type string.
+func getTemplateFromFile(context *context.Context) (string, string) {
+
+	rawTemplate, readFileError := ioutil.ReadFile(*context.CliArguments.TemplatePath)
+	if readFileError != nil {
+		context.Logger.Error(readFileError.Error())
+	}
+	rawStackName := *context.CliArguments.Stack
+	template := string(rawTemplate)
+	stackName := string(rawStackName)
+	return template, stackName
+}
+
+// This function uses CreateStackInput variable to create Stack.
+func createStack(templateStruct cloudformation.CreateStackInput, session *session.Session) {
+	api := cloudformation.New(session)
+	api.CreateStack(&templateStruct)
+}
+
+//This function uses all functions above and session to create Stack.
+func NewStack(context *context.Context) {
+	tokenError := updateSessionToken(context.Config.DefaultProfile, context.Config.DefaultRegion, context.Config.DefaultDurationForMFA, context)
+	if tokenError != nil {
+		context.Logger.Error(tokenError.Error())
+	}
+	session, createSessionError := createSession(context, context.Config.DefaultProfile, &context.Config.DefaultRegion)
+	if createSessionError != nil {
+		context.Logger.Error(createSessionError.Error())
+	}
+	template, stackName := getTemplateFromFile(context)
+	templateStruct := createStackInput(context, &template, &stackName)
+
+	createStack(templateStruct, session)
+}
+
+//This function bases on "DeleteStackInput" structure and destroys stack. It uses "StackName" to choose which stack will be destroy. Before that it creates session.
+func DestroyStack(context *context.Context) {
+	delStackInput := deleteStackInput(context)
+	session, sessionError := createSession(context, context.Config.DefaultProfile, &context.Config.DefaultRegion)
+	if sessionError != nil {
+		context.Logger.Error(sessionError.Error())
+	}
+	api := cloudformation.New(session)
+	api.DeleteStack(&delStackInput)
+}
+
+//This function gets "StackName" from Stack in CliArguments and creates "DeleteStackInput" structure.
+func deleteStackInput(context *context.Context) cloudformation.DeleteStackInput {
+	name := *context.CliArguments.Stack
+	templateStruct := cloudformation.DeleteStackInput{
+		StackName: &name,
+	}
+	return templateStruct
+}
+
+//"createSession" and "updateSessionToken" are from onlinevalidator.go file. They allow to connect with AWS.
```

Comment:
You can reuse `createSession` and `updateSessionToken` from onlinevalidator instead of copying the code. Create new module with these functions and use it in both places.

Reply:
Functions should be public in this new module?

Reply:
Yes 👍